### DISCUSSION TITLE
How can I call/materialize a `graph_asset` within another `graph_asset`?

### DISCUSSION THREAD

**Discussion by pranavm716** [dagster-io/dagster#17742] [Q&A] [answered]
Hi team, 

I want to I call/materialize a `graph_asset` within another `graph_asset`. This is my minimal example:

```python3
from dagster import graph_asset, op


@op
def op_1():
    return 1


@graph_asset
def asset_1() -> int:
    return op_1()


@graph_asset
def asset_2() -> int:
    result = asset_1()
    return result * 2

```

I want to get the result of calling `asset_1()` inside `asset_2`, perform some transformations to it (multiplying by 2 in this example), then return the transformed result. However, when I try to launch the dagster UI, I get this error:

```
TypeError: unsupported operand type(s) for *: 'InvokedNodeOutputHandle' and 'int'
```

Any help is appreciated, thanks!

**Comment by jamiedemaria** [comment, marked as answer]
Hey @pranavm716 . `@graph_asset`s are a way to set the dependency structure of your ops/assets, but they can't do computation. If some more technical detail is helpful, when dagster first starts running, it functions decorated by `@graph_asset` get executed to determine the structure of the graph, that's why there can't be any computation in them. 

instead you'll want to do something like this 
```python
@op
def op_1():
    return 1

@graph_asset
def asset_1() -> int:
    return op_1()

@op 
def multiply_by_2(x):
    return x * 2

@graph_asset
def asset_2(asset_1) -> int:
    return multiply_by_2(asset_1)

```
